src/ui/pages/zcl_abapgit_gui_page_diff.clas.abap

```diff
@@ -1,11 +1,16 @@
 CLASS zcl_abapgit_gui_page_diff DEFINITION
   PUBLIC
-  INHERITING FROM zcl_abapgit_gui_page
-  CREATE PUBLIC.
+  INHERITING FROM zcl_abapgit_gui_component
+  CREATE PRIVATE
+  GLOBAL FRIENDS zcl_abapgit_gui_page_patch.
```

Comment:
this new FRIENDS looks suspicions?

Reply:
Yeah, the UI framework isn't made for pages inheriting from other pages. I will re-do it with one super and diff and patch as children (like code inspector + syntax check)

Reply:
does it need to be inheritance? can it be done via composition? ie. extract some logic to a separate class